src/redisearch_rs/rlookup/tests/row.rs

```diff
@@ -0,0 +1,214 @@
+
+    // insert a key at the second position
+    let mut key = RLookupKey::new(c"test2", RLookupKeyFlags::empty());
+    key.dstidx = 1;
+    row.write_key(&key, MockRSValue::create_num(84.0));
+    assert!(!row.is_empty());
+    assert_eq!(row.len(), 2);
+    assert_eq!(row.num_dyn_values(), 2);
+    assert_eq!(row.dyn_values()[1].as_ref().unwrap().as_num(), Some(84.0));
+}
+
+#[test]
+fn test_insert_with_gap() {
+    let mut row: RLookupRow<MockRSValue> = RLookupRow::new();
+    assert!(row.is_empty());
+    assert_eq!(row.len(), 0);
+    assert_eq!(row.num_dyn_values(), 0);
+
+    // generate test key at index 15
+    let mut key = RLookupKey::new(c"test", RLookupKeyFlags::empty());
+    key.dstidx = 15;
+    row.write_key(&key, MockRSValue::create_num(42.0));
+
+    assert!(!row.is_empty());
+    assert_eq!(row.len(), 16); // Length should be 16 due to the gap
+    assert_eq!(row.num_dyn_values(), 1);
+    assert_eq!(row.dyn_values()[15].as_ref().unwrap().as_num(), Some(42.0));
+}
+
+#[test]
+fn test_insert_non_owned() {
+    let mut row: RLookupRow<MockRSValue> = RLookupRow::new();
+    assert!(row.is_empty());
+    assert_eq!(row.len(), 0);
+    assert_eq!(row.num_dyn_values(), 0);
+
+    // generate test key at index 0
+    let key = RLookupKey::new(c"test", RLookupKeyFlags::empty());
+
+    // insert a key at the first position
+    let mock = MockRSValue::create_num(42.0);
+    row.write_key(&key, mock.clone());
+
+    // We have the key outside of the row, so it should have a ref count of 2
+    assert_eq!(
+        Rc::strong_count(&row.dyn_values()[0].as_ref().unwrap().value),
+        2
+    );
+
+    drop(mock);
+    // After dropping, the ref count should be back to 1
+    assert_eq!(
+        Rc::strong_count(&row.dyn_values()[0].as_ref().unwrap().value),
+        1
+    );
+}
+
+#[test]
+fn test_insert_overwrite() {
+    let mut row: RLookupRow<MockRSValue> = RLookupRow::new();
+    assert!(row.is_empty());
+    assert_eq!(row.len(), 0);
+    assert_eq!(row.num_dyn_values(), 0);
+
+    // generate test key at index 0
+    let key = RLookupKey::new(c"test", RLookupKeyFlags::empty());
+
+    // insert a key at the first position
+    let mock_to_be_overwritten = MockRSValue::create_num(42.0);
+    row.write_key(&key, mock_to_be_overwritten.clone());
+    assert_eq!(Rc::strong_count(&mock_to_be_overwritten.value), 2);
+
+    assert!(!row.is_empty());
+    assert_eq!(row.len(), 1);
+    assert_eq!(row.num_dyn_values(), 1);
+    assert_eq!(row.dyn_values()[0].as_ref().unwrap().as_num(), Some(42.0));
+
+    // overwrite the value at the same index
+    row.write_key(&key, MockRSValue::create_num(84.0));
+    assert_eq!(row.num_dyn_values(), 1);
+    assert_eq!(row.dyn_values()[0].as_ref().unwrap().as_num(), Some(84.0));
+    // The overwritten value should have been decremented
+    assert_eq!(
+        Rc::strong_count(&row.dyn_values()[0].as_ref().unwrap().value),
+        1
+    );
+}
+
+#[test]
+fn test_wipe() {
```

Comment:
Do you think we should test writing again after the wiping?
Or is this test enough?